code/modules/power/supermatter/supermatter_extra_effects.dm

```diff
@@ -130,9 +130,10 @@
 	if(zap_count >= 1)
 		playsound(loc, 'sound/weapons/emitter2.ogg', 100, TRUE, extrarange = 10)
 		var/delta_time = (SSmachines.times_fired - last_high_energy_zap_perspective_machines) * SSmachines.wait / (1 SECONDS)
-		for(var/i in 1 to zap_count)
-			supermatter_zap(src, range, clamp(internal_energy * 3200, 6.4e6, 3.2e7) * delta_time, flags, zap_cutoff = src.zap_cutoff * delta_time, power_level = internal_energy, zap_icon = src.zap_icon)
-		last_high_energy_zap_perspective_machines = SSmachines.times_fired
+		if(delta_time)
+			for(var/i in 1 to zap_count)
+				supermatter_zap(src, range, clamp(internal_energy * 3200, 6.4e6, 3.2e7) * delta_time, flags, zap_cutoff = src.zap_cutoff * delta_time, power_level = internal_energy, zap_icon = src.zap_icon)
+			last_high_energy_zap_perspective_machines = SSmachines.times_fired
```

Comment:
this means if atmos is ticking fast we "spend" heat and potentially integrity fast and don't gain consistent power from it. is that ok?

Reply:
The power will be consistent with SSmachines time perspective, which is what probably more than 99% of the power consumption in a round will use. It will be inconsistent when using a different time perspective, however the supermatter is pretty much just a source of power coming out of thin air, and doesn't require any resources or entropy to gain energy from it.